wicker/plugins/wandb.py

```diff
@@ -0,0 +1,95 @@
+import os
+from typing import Any, Dict, Literal
+
+import wandb
+
+from wicker.core.config import get_config
+from wicker.core.definitions import DatasetID
+from wicker.core.storage import S3PathFactory
+
+
+def version_dataset(
+    dataset_name: str,
+    dataset_version: str,
+    metadata: Dict[str, Any],
+    dataset_backend: Literal["s3"] = "s3",
+) -> None:
+    """
+    Version the dataset on Weights and Biases using the config parameters defined in wickerconfig.json.
+
+    Args:
+        dataset_name: The name of the dataset to be versioned
+        dataset_version: The version of the dataset to be versioned
+        metadata: The metadata to be logged as an artifact, enforces dataclass for metadata documentation
+        dataset_backend: The backend where the dataset is stored, currently only supports s3
+    """
+    # needs to first acquire and set wandb creds
+    # WANDB_API_KEY, WANDB_BASE_URL, WANDB_USER_EMAIL
```

Comment:
Do we need user email here?

Reply:
No we don't, I thought we did but we don't, removed.